addons/test_mail/tests/test_mail_mail.py

```diff
@@ -25,6 +26,43 @@ def test_mail_message_notify_from_mail_mail(self):
         self.assertSentEmail(mail.env.user.partner_id, ['test@example.com'])
         self.assertEqual(len(self._mails), 1)
 
+    def test_mail_mail_sending_batch(self):
+        """Test that the mails are send in batch.
+
+        Batch are defined by the mail server and the email from field.
+        """
+        mail_server = self.env['ir.mail_server'].create({'name': 'server', 'smtp_host': 'qsd'})
+        IrMailServer = type(self.env['ir.mail_server'])
+        find_mail_server = self.env['ir.mail_server']._find_mail_server
+        connect = self.env['ir.mail_server'].connect
+
+        mails = self.env['mail.mail'].create([{
+            'body_html': '<p>Test</p>',
+            'email_to': 'test@example.com',
+            'email_from': 'test@example.com',
+            'partner_ids': [(4, self.user_employee.partner_id.id)]
+        } for _ in range(5)]) | self.env['mail.mail'].create([{
+            'body_html': '<p>Test</p>',
+            'email_to': 'test@odoo.com',
+            'email_from': 'test@odoo.com',
+            'partner_ids': [(4, self.user_employee.partner_id.id)]
+        } for _ in range(5)]) | self.env['mail.mail'].create([{
+            'body_html': '<p>Test</p>',
+            'email_to': 'test@odoo.com',
+            'email_from': 'test@odoo.com',
+            'partner_ids': [(4, self.user_employee.partner_id.id)],
+            'mail_server_id': mail_server.id,
+        } for _ in range(5)])
+
+        with patch.object(TestingSMTPSession, 'send_email') as patched_send_email,\
```

Comment:
To check probably later once all code is validated, but I would like those to be available directly when using "with self.mock_mail_gateway()" to avoid redefining yet another patch/mock. I feel like we could re-use existing mocks or at least see what is missing or badly use to avoid using custom test classes and patchs.

Reply:
The thing here is that we need to have the patched methods to be able to know how many time they have been called, with which parameters, etc... I do not think we will need this for tests in sub mail module, no ?

(if yes, we can still add attribute to the `MockEmail` with the patched methods, send_mail, _find_mail_server, connect,... but it will complexify the code for a specific usage)

Reply:
So I made my own mock method